tests/cgal/cgal_mesh_criteria.cc

```diff
@@ -0,0 +1,76 @@
+// ---------------------------------------------------------------------
+//
+// Copyright (C) 2022 by the deal.II authors
+//
+// This file is part of the deal.II library.
+//
+// The deal.II library is free software; you can use it, redistribute
+// it, and/or modify it under the terms of the GNU Lesser General
+// Public License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+// The full text of the license can be found in the file LICENSE.md at
+// the top level directory of deal.II.
+//
+// ---------------------------------------------------------------------
+
+// Check that different `cell_size` parameter gives different number of cells.
+
+#include <deal.II/base/config.h>
+
+#include <deal.II/grid/grid_generator.h>
+#include <deal.II/grid/tria.h>
+
+#include <CGAL/IO/File_medit.h>
+#include <CGAL/IO/io.h>
+#include <deal.II/cgal/utilities.h>
+
+#include "../tests.h"
+
+using K         = CGAL::Exact_predicates_inexact_constructions_kernel;
+using CGALPoint = CGAL::Point_3<K>;
+using namespace CGALWrappers;
+using Mesh_domain =
+  CGAL::Polyhedral_mesh_domain_with_features_3<K,
+                                               CGAL::Surface_mesh<CGALPoint>>;
+using Tr = typename CGAL::
+  Mesh_triangulation_3<Mesh_domain, CGAL::Default, ConcurrencyTag>::type;
+using Mesh_criteria = CGAL::Mesh_criteria_3<Tr>;
+using C3t3          = CGAL::Mesh_complex_3_in_triangulation_3<Tr,
+                                                     Mesh_domain::Corner_index,
+                                                     Mesh_domain::Curve_index>;
+
+
+void
+test()
+{
+  CGAL::Surface_mesh<CGALPoint> sm;
+  C3t3                          tria;
+  std::ifstream                 input("input_grids/cube.off");
+  input >> sm;
+  cgal_surface_mesh_to_cgal_triangulation(sm,
+                                          tria,
+                                          CGAL::parameters::cell_size = 0.1);
+
+  deallog << "Number of cells: " << tria.number_of_facets_in_complex()
+          << std::endl;
+  std::ofstream first_output_medit("first_cube.mesh");
+  tria.output_to_medit(first_output_medit, false);
+
+  tria.clear();
+
+  cgal_surface_mesh_to_cgal_triangulation(sm,
+                                          tria,
+                                          CGAL::parameters::cell_size = 0.5);
+
+  deallog << "Number of cells: " << tria.number_of_facets_in_complex()
+          << std::endl;
+  std::ofstream second_output_medit("second_cube.mesh");
+  tria.output_to_medit(second_output_medit, false);
+}
```

Comment:
Are you sure that all versions of CGAL give the same result? I have seen different results. I think here it would be just safe to check that the number of facets in the first case is larger than in the second case. Also, remove all intermediate files with `remove("second_cube.mesh")`, or just remove the writing of it. This does not test anything deal.II related, so if you checked that visually everything is fine, there is no need to write it out using CGAL functions.